esp-wifi/src/wifi/mod.rs

```diff
@@ -2356,6 +2484,140 @@ fn apply_sta_config(config: &ClientConfiguration) -> Result<(), WifiError> {
+                identity.as_str().as_ptr(),
+                identity.len() as i32
+            ))?;
+        } else {
+            esp_eap_client_clear_identity();
+        }
+
+        if let Some(username) = &config.username {
+            esp_wifi_result!(esp_eap_client_set_username(
+                username.as_str().as_ptr(),
+                username.len() as i32
+            ))?;
+        } else {
+            esp_eap_client_clear_username();
+        }
+
+        if let Some(password) = &config.password {
+            esp_wifi_result!(esp_eap_client_set_password(
+                password.as_str().as_ptr(),
+                password.len() as i32
+            ))?;
+        } else {
+            esp_eap_client_clear_password();
+        }
+
+        if let Some(new_password) = &config.new_password {
+            esp_wifi_result!(esp_eap_client_set_new_password(
+                new_password.as_str().as_ptr(),
+                new_password.len() as i32
+            ))?;
+        } else {
+            esp_eap_client_clear_new_password();
+        }
+
+        if let Some(pac_file) = &config.pac_file {
+            esp_wifi_result!(esp_eap_client_set_pac_file(
+                pac_file.as_ptr(),
+                pac_file.len() as i32
+            ))?;
+        }
+
+        if let Some(phase2_method) = &config.ttls_phase2_method {
+            esp_wifi_result!(esp_eap_client_set_ttls_phase2_method(
+                phase2_method.to_raw()
+            ))?;
+        }
+
+        if let Some(ca_cert) = config.ca_cert {
+            esp_wifi_result!(esp_eap_client_set_ca_cert(
+                ca_cert.as_ptr(),
+                ca_cert.len() as i32
+            ))?;
+        } else {
+            esp_eap_client_clear_ca_cert();
+        }
+
+        if let Some((cert, key, password)) = config.certificate_and_key {
+            let (pwd, pwd_len) = if let Some(pwd) = password {
+                (pwd.as_ptr(), pwd.len() as i32)
+            } else {
+                (core::ptr::null(), 0)
+            };
+
+            esp_wifi_result!(esp_eap_client_set_certificate_and_key(
+                cert.as_ptr(),
+                cert.len() as i32,
+                key.as_ptr(),
+                key.len() as i32,
+                pwd,
+                pwd_len,
+            ))?;
+        } else {
+            esp_eap_client_clear_certificate_and_key();
+        }
+
+        if let Some(cfg) = &config.eap_fast_config {
+            let params = esp_eap_fast_config {
+                fast_provisioning: cfg.fast_provisioning as i32,
+                fast_max_pac_list_len: cfg.fast_max_pac_list_len as i32,
+                fast_pac_format_binary: cfg.fast_pac_format_binary,
+            };
+            esp_wifi_result!(esp_eap_client_set_fast_params(params))?;
+        }
+
+        esp_wifi_result!(esp_eap_client_set_disable_time_check(!&config.time_check))?;
+
+        // esp_eap_client_set_suiteb_192bit_certification unsupported because we build
```

Comment:
Do we need to error out if these are set in the config?

Reply:
Currently we don't offer a way to ask for these two settings